ApiData.ts

```diff
@@ -0,0 +1,201 @@
+class ApiData {
+	// Properties to manage data and settings
+	pageSize: number;
+	currentPage = 1;
+	data: GridData[] = [];
+	totalItems = 0;
+	columnNames: ColumnName[] = [];
+	maxGridHeight = 0;
+	firstVal = 0;
+	lastVal: number | undefined;
+
+	constructor(pageSize: number) {
+		this.pageSize = pageSize;
+	};
+	// Initialize method to set up the grid
```

Comment:
Please change this to the javadoc format. This is unfortunately not in our style guide :( But it really should be.
```suggestion
	/** Initialize method to set up the grid */
```